buildSrc/src/main/groovy/org/apache/beam/gradle/VendorJavaPlugin.groovy

```diff
@@ -286,8 +288,23 @@ artifactId=${project.name}
               pom.withXml {
                 def root = asNode()
                 def dependenciesNode = root.appendNode('dependencies')
+                def unshaded = { dependency ->
+                  if (dependency instanceof ProjectDependency) {
+                    def groupId = dependency.getDependencyProject().mavenGroupId
+                    def artifactId = dependency.getDependencyProject().archivesBaseName
+                    def version = dependency.version
+                    return config.unshadedDependencies != null && config.unshadedDependencies.contains("$groupId:$artifactId:$version")
+                  } else {
+                    def groupId = dependency.group
+                    def artifactId = dependency.name
+                    def version = dependency.version
+                    return config.unshadedDependencies != null && config.unshadedDependencies.contains("$groupId:$artifactId:$version")
```

Comment:
This does not include the classifier. And since leaving out the classifier is equal to `jar` (I think always?) it is not just a string comparison.

Does Gradle utilities have anything to parse the `unshadedDependencies` into `ProjectDependency` objects and then use equality on that object?